core/src/test/java/org/apache/brooklyn/entity/group/DynamicRegionsFabricTest.java

```diff
@@ -60,16 +64,98 @@ public void setUp() throws Exception {
     }
 
     @Test
-    public void testUsesInitialLocations() throws Exception {
+    public void testUsesInitialLocationsFromStartEffectorArgs() throws Exception {
         app.start(ImmutableList.of(loc1, loc2));
 
-        assertEquals(fabric.getChildren().size(), 2, "children="+fabric.getChildren());
-        assertEquals(fabric.getMembers().size(), 2, "members="+fabric.getMembers());
-        assertEqualsIgnoringOrder(fabric.getChildren(), fabric.getMembers());
-        assertEqualsIgnoringOrder(getLocationsOfChildren(fabric), ImmutableList.of(loc1, loc2));
+        assertFabricChildren(fabric, 2, ImmutableList.of(loc1, loc2));
+    }
+    
+    @Test
+    @SuppressWarnings("deprecation")
+    public void testUsesInitialLocationsFromAppSpec() throws Exception {
+        TestApplication app2 = mgmt.getEntityManager().createEntity(EntitySpec.create(TestApplication.class)
+            .configure(BrooklynConfigKeys.SKIP_ON_BOX_BASE_DIR_RESOLUTION, true)
+            .child(EntitySpec.create(DynamicRegionsFabric.class)
+                .configure("memberSpec", EntitySpec.create(TestEntity.class)))
+            .locations(ImmutableList.of(loc1, loc2)));
+        DynamicRegionsFabric fabric2 = (DynamicRegionsFabric) Iterables.getOnlyElement(app2.getChildren());
+        app2.start(ImmutableList.<Location>of());
+        
+        assertFabricChildren(fabric2, 2, ImmutableList.of(loc1, loc2));
     }
     
     @Test
+    @SuppressWarnings("deprecation")
+    public void testUsesInitialLocationsFromEntitySpec() throws Exception {
```

Comment:
Not quite the same. In `testUsesInitialLocationsFromAppSpec` we set the locations on the app's spec. In `testUsesInitialLocationsFromEntitySpec` we set the locations on the fabric's spec.

Could refactor to remove duplication, but I think that it's probably simpler to have the 7 lines duplicated rather than a delegate method with an if-else block (given setting the location is buried in the 5 lines required to build the spec).